dockerbuilder/defaultdockerbuilder/dockerbuilder.go

```diff
@@ -60,8 +94,127 @@ func (d *DefaultDockerBuilder) RunDockerBuild(dockerID distgo.DockerID, productT
 		}
 		args = append(args, buildArgsFromScript...)
 	}
-	args = append(args, contextDirPath)
 
-	cmd := exec.Command("docker", args...)
-	return distgo.RunCommandWithVerboseOption(cmd, verbose, dryRun, stdout)
+	if d.OutputType&allOutputs == 0 {
+		return errors.New("A valid output type of docker builder must be specified")
+	}
+
+	if d.OutputType&OCILayout != 0 {
+		if err := d.ensureDockerContainerDriver(dockerID, verbose, dryRun, stdout); err != nil {
+			return err
+		}
+		destDir := productTaskOutputInfo.ProductDockerOCIDistOutputDir(dockerID)
+		destFile := fmt.Sprintf("%s/image.tar", destDir)
+
+		ociArgs := append([]string{"buildx"}, args...)
+		ociArgs = append(ociArgs, d.BuildxPlatformArg, fmt.Sprintf("--output=type=oci,dest=%s", destFile), contextDirPath)
+		cmd := exec.Command("docker", ociArgs...)
+		if err := distgo.RunCommandWithVerboseOption(cmd, verbose, dryRun, stdout); err != nil {
+			return err
+		}
+		if !dryRun {
+			if err := d.extractToOCILayout(destDir, destFile); err != nil {
+				return err
+			}
+		}
+	}
+	if d.OutputType&DockerDaemon != 0 {
+		cmd := exec.Command("docker", append(args, contextDirPath)...)
+		if err := distgo.RunCommandWithVerboseOption(cmd, verbose, dryRun, stdout); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// extractToOCILayout is responsible for converting the buildx OCI tarball output to a compatible OCI layout on disk.
+// The buildx tarball adds a layer of indirection which doesn't seem to play nicely with some registries; the top-level
+// image index produced contains a manifest per-tag, which point to the "actual" image index we want to publish. Since
+// we know all the rendered tags at publish time, we can move the "actual" image index back to the top level and do a
+// publish per-tag.
+func (d *DefaultDockerBuilder) extractToOCILayout(destOCILayoutDir, sourceOCITarball string) error {
+	if err := archiver.DefaultTar.Unarchive(destOCILayoutDir, sourceOCITarball); err != nil {
```

Comment:
Looks like this is the wrong way round, should be source tarball and the directory to extract it into
```suggestion
	if err := archiver.DefaultTar.Unarchive(sourceOCITarball, destOCILayoutDir); err != nil {
```

Reply:
Good catch! Updated